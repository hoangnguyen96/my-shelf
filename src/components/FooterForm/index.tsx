import { Box, StyleProps, Text } from "@chakra-ui/react";
import Link from "next/link";
import { memo } from "react";

interface FooterFormProps extends StyleProps {
  text: string;
  link?: string;
  textLink: string;
}
const FooterForm = ({
  text,
  link = "",
  textLink,
  ...rest
}: FooterFormProps) => (
  <Box pb="160px" {...rest}>
    <Text as="span">
      {text}{" "}
      <Link href={link} style={{ textDecoration: "underline" }}>
        {textLink}
      </Link>
    </Text>
  </Box>
);

export default memo(FooterForm);
