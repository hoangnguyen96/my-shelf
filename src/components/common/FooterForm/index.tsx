import { Box, StyleProps, Text } from "@chakra-ui/react";
import Link from "next/link";

interface FooterFormProps extends StyleProps {
  text: string;
  link: string;
  textLink: string;
}
const FooterForm = ({ text, link, textLink, ...rest }: FooterFormProps) => (
  <Box pb="160px" {...rest}>
    <Text as="span">
      {text}{" "}
      <Link href={link} style={{ textDecoration: "underline" }}>
        {textLink}
      </Link>
    </Text>
  </Box>
);

export default FooterForm;
