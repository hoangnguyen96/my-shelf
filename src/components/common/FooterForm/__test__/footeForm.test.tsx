import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import FooterForm from "..";

describe("FooterForm", () => {
  it("Should render correctly snapshot", () => {
    expect(
      render(<FooterForm text="Name" textLink="Login" />)
    ).toMatchSnapshot();
  });
});
