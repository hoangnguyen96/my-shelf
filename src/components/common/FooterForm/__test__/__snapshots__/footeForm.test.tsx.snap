// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterForm Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-15286ub"
      >
        <span
          class="chakra-text css-0"
        >
          Name
           
          <a
            href="/"
            style="text-decoration: underline;"
          >
            Login
          </a>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-15286ub"
    >
      <span
        class="chakra-text css-0"
      >
        Name
         
        <a
          href="/"
          style="text-decoration: underline;"
        >
          Login
        </a>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
